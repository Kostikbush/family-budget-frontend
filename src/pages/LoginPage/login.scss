@import "../../Common-Style/mixins.scss";
@import "../../Common-Style/colors.scss";
.app__login-page {
  @include bgPage();
}

.app-login-page-forms__form {
  color: white;
  letter-spacing: 2px;
  font-size: 20px;
  height: 100%;
  display: flex;
  flex-direction: column;
  justify-content: space-between;
  align-items: center;
  & h2 {
    padding: 0 0 5px 0;
  }
}
.app-login-page-forms-form__btns {
  display: flex;
  width: 100%;
  justify-content: space-between;
}
.app-login-page-forms-form__inputs {
  height: 60%;
  width: 100%;
  display: flex;
  flex-direction: column;
  justify-content: space-around;
}
.app-login-page-forms-form-registration-input {
  height: 60%;
  width: 100%;
  display: flex;
  flex-direction: column;
  justify-content: space-around;
}

.form-change-enter {
  transform: translateX(-110%);
}
.form-change-enter-active {
  transform: translateX(0%);
}
.form-change-exit {
  transform: translateX(0%);
}

.form-change-exit-active {
  transform: translateX(110%);
}
.form-change-enter-active,
.form-change-exit-active {
  transition: all 300ms;
}

.app-login-page-forms {
  height: 100vh;
  display: flex;
  align-items: center;
  justify-content: center;
}
.app-error-login {
  width: 100%;
  height: 45%;
  display: flex;
  flex-direction: column;
  justify-content: space-around;
  & li {
    display: flex;
    align-items: center;
    justify-content: center;
    height: auto;
    font-size: 15px;
    line-height: 18px;
    color: white;
    padding: 10px;
    @include borderEl();
    //@include shadowErr();
    background-color: $bgElNoTransparent;
    & span {
      margin-right: 10px;
      font-size: 15px;
    }
    & .al-self-icon {
      color: $colorTextError;
    }
  }
}
.page-blur {
  filter: blur(2px);
}
.app-error-login__error-reg {
  display: flex;
  flex-direction: column;
  justify-content: space-between;
}
.app-error-login-error-reg__errors-list {
  display: flex;
  flex-direction: column;
}

.change-form-loginPgae-enter {
  //opacity: 0;
  transform: translateX(-110%);
}
.change-form-loginPgae-enter-active {
  //opacity: 1;
  transform: translateX(0%);
}
.change-form-loginPgae-exit {
  //opacity: 1;
  transform: translateX(0%);
}
.change-form-loginPgae-exit-active {
  //opacity: 0;
  transform: translateX(110%);
}
.change-form-loginPgae-enter-active,
.change-form-loginPgae-exit-active {
  transition: transform 400ms;
}
@media (max-width: 470px) {
  .app-login-page-forms__form {
    & h2 {
      font-size: 15px;
    }
  }
  .app-login-page-forms-form__btns {
    height: 30%;
    width: 100%;
    align-items: center;
    flex-direction: column;
  }
}
@media (max-height: 400px) {
  .app-error-login {
    & li {
      padding: 5px;
    }
  }
}
@media (max-width: 350px) {
  .app-error-login {
    & li {
      padding: 0px;
      & span {
        font-size: 10px;
      }
    }
  }
}
