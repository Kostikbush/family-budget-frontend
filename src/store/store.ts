import { configureStore } from "@reduxjs/toolkit";
import { authApi } from "../service/authApi";
import { budgetApi } from "../service/budgetApi";
import chatDataSlice from "./reducers/ChatData";
import userAftorasationSlice from "./reducers/UserAftorasationSlice";

export const store = configureStore({
  reducer: {
    ayth: userAftorasationSlice,
    chat: chatDataSlice,
    [authApi.reducerPath]: authApi.reducer,
    [budgetApi.reducerPath]: budgetApi.reducer,
  },
  middleware: (getDefaultMiddleware) =>
    getDefaultMiddleware().concat(authApi.middleware, budgetApi.middleware),
});

export type RootState = ReturnType<typeof store.getState>;
// Inferred type: {posts: PostsState, comments: CommentsState, users: UsersState}
export type AppDispatch = typeof store.dispatch;
