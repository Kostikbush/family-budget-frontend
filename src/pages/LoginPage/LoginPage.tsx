import { useEffect, useState } from "react";
import { SwitchTransition, CSSTransition } from "react-transition-group";

import { Alert } from "../../UI/Alert/Alert";

import { useAppDispatch } from "../../hooks/redux";
import { dataUserSaveStore } from "../../store/reducers/UserAftorasationSlice";

import "./login.scss";
import { getDataFromIndexDB } from "../../helpersFunc/saved";
import { changeBody } from "../../helpersFunc/changeBody";
import { ChangeBgPages } from "../../CONST/ChangeBgPages";

import { checkOnlineStatus } from "../../helpersFunc/checkOnline";
import { RegistrationForm } from "./RegistrashionForm/RegistrationForm";
import { LoginForm } from "./LoginForm/LoginForm";
import { useCheckBackMutation } from "../../service/authApi";
import { QueryResult } from "../../models/typeQueryResult";
import { ErrorBackWork } from "../../models/IErrorBackend";
import { LoaderPage } from "../../UI/LoaderPage/LoaderPage";
import { WrapperForm } from "../../UI/WrapperForm/WrapperForm";

export const LoginPage = () => {
  const [errorsValidations, setErrorsValidations] = useState<string[]>([]);
  const [errorReg, seteErrorReg] = useState(undefined);
  const [errorLog, seteErrorLog] = useState(undefined);
  const [isVieEnterForm, setIsVieEnterForm] = useState(true);
  const [isOffline, setIsOffline] = useState(false);
  const [checkBack, { error, isLoading, isError }] =
    useCheckBackMutation<QueryResult<any, ErrorBackWork>>();
  ////////////////////////////===================================

  /////////////////////////////==================================
  const dispatch = useAppDispatch();
  useEffect(() => {
    getDataFromIndexDB(dispatch, dataUserSaveStore);
    changeBody(ChangeBgPages.LOGIN);
    checkOnlineStatus(setIsOffline);
    checkBack();
    //eslint-disable-next-line react-hooks/exhaustive-deps
  }, []);
  const handleClickChangeForm = () => {
    setIsVieEnterForm(!isVieEnterForm);
  };

  return (
    <>
      <main className="app__login-page page-bg-move">
        {isLoading && <LoaderPage />}
        {!isError && !isLoading && (
          <section className="container">
            <div className="app-login-page-forms">
              <WrapperForm errorValidation={errorsValidations.length > 0}>
                <SwitchTransition mode="out-in">
                  <CSSTransition
                    timeout={300}
                    classNames="change-form-loginPgae"
                    key={isVieEnterForm ? "1" : "2"}
                    in={isVieEnterForm}
                  >
                    {isVieEnterForm ? (
                      <LoginForm
                        seteErrorLog={seteErrorLog}
                        setErrorsValidations={setErrorsValidations}
                        handleClickChangeForm={handleClickChangeForm}
                      />
                    ) : (
                      <RegistrationForm
                        seteErrorReg={seteErrorReg}
                        setErrorsValidations={setErrorsValidations}
                        handleClickChangeForm={handleClickChangeForm}
                      />
                    )}
                  </CSSTransition>
                </SwitchTransition>
              </WrapperForm>
            </div>
          </section>
        )}
      </main>
      {/* <Alert
        isError={isOffline}
        handleBooleanValue={setIsOffline}
        message="У вас нет поключения к интернету"
        type="error"
      /> */}
      <Alert
        errorWorkBack={error}
        isError={isOffline}
        message={`Сейчас бэкенд этого сервиса не работает :(. Мы уже работает над этим. Попробуйте зайти позже. Так же проверте подключение к интернету.`}
        type="error"
      />
      <Alert
        message="У вас есть неправильные поля"
        type="error"
        errorsValidtionForm={errorsValidations}
        setRessetArray={setErrorsValidations}
      />
      <Alert
        message="У вас есть неправильные поля со значениями"
        type="error"
        errorsFromBack={errorReg}
      />
      <Alert type="error" errorsFromBack={errorLog} />
    </>
  );
};
