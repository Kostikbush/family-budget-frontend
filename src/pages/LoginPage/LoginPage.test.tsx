/* eslint-disable testing-library/no-unnecessary-act */
import "@testing-library/jest-dom/extend-expect";
import { act, render, screen } from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import "fake-indexeddb/auto";
import { Provider } from "react-redux";
import { Route, Routes, BrowserRouter } from "react-router-dom";
import { store } from "../../store/store";
import { LoginPage } from "./LoginPage";
describe("Test Login Page", () => {
  test("LoginPage vie", async () => {
    render(
      <Provider store={store}>
        <BrowserRouter>
          <Routes>
            <Route path="/" element={<LoginPage />} />
          </Routes>
        </BrowserRouter>
      </Provider>
    );
    jest.setTimeout(500);
    const input = await screen.findByPlaceholderText(/Введите email/i);
    const enter = await screen.findByText(/вход/i);
    expect(enter).toBeInTheDocument();
    expect(input).toBeInTheDocument();
    const formEnter = screen.getByTestId("form-enter-snap");
    expect(formEnter).toBeInTheDocument();
    expect(formEnter).toMatchSnapshot();
  });
  test("Click to btn change form", async () => {
    act(() => {
      render(
        <Provider store={store}>
          <BrowserRouter>
            <Routes>
              <Route path="/" element={<LoginPage />} />
            </Routes>
          </BrowserRouter>
        </Provider>
      );
    });
    jest.setTimeout(500);
    const btnChangeForm = await screen.findByTestId("btn-testCgangeForm");
    const inputForm = screen.getByPlaceholderText(/Введите email/i);
    await act(async () => {
      await userEvent.type(inputForm, "Konstantin");
    });
    expect(btnChangeForm).toBeInTheDocument();
    expect(screen.getByPlaceholderText("Введите email")).toContainHTML(
      "Konstantin"
    );
    await act(async () => {
      await userEvent.click(screen.getByTestId("btn-testCgangeForm"));
    });
    const newH2 = await screen.findByTestId(/h2-registration/i);
    expect(newH2).toContainHTML("Регистрация");
  });
});
