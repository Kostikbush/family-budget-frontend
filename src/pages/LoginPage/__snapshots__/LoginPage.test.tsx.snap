// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Login Page LoginPage vie 1`] = `
<form
  class="app-login-page-forms__form"
  data-testid="form-enter-snap"
>
  <h2>
    Вход
  </h2>
  <div
    class="app-login-page-forms-form-registration-input"
  >
    <div
      class="input-wrapper input-wrapper-color-login"
    >
      <input
        class="input-wrapper__input input-color-login"
        placeholder="Введите email"
        type="text"
        value=""
      />
      <span
        class="input-wrapper-btn__stroke"
      />
    </div>
    <div
      class="input-wrapper input-wrapper-color-login"
    >
      <input
        class="input-wrapper__input input-color-login"
        placeholder="Введите пароль"
        type="password"
        value=""
      />
      <button
        class="input-wrapper__btn"
      >
        <svg
          fill="currentColor"
          height="17"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 1024 1024"
          width="17"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M396 512a112 112 0 1 0 224 0 112 112 0 1 0-224 0zm546.2-25.8C847.4 286.5 704.1 186 512 186c-192.2 0-335.4 100.5-430.2 300.3a60.3 60.3 0 0 0 0 51.5C176.6 737.5 319.9 838 512 838c192.2 0 335.4-100.5 430.2-300.3 7.7-16.2 7.7-35 0-51.5zM508 688c-97.2 0-176-78.8-176-176s78.8-176 176-176 176 78.8 176 176-78.8 176-176 176z"
          />
        </svg>
      </button>
      <span
        class="input-wrapper-btn__stroke"
      />
    </div>
  </div>
  <div
    class="app-login-page-forms-form__btns"
  >
    <button
      class="btn border-pinc"
      data-testid="button-enter"
      type="submit"
    >
      Войти
      <span
        class="btn__stroke"
      />
    </button>
    <button
      class="btn border-pinc"
      data-testid="btn-testCgangeForm"
    >
      Регистрация
      <span
        class="btn__stroke"
      />
      <svg
        class="arr-ml"
        fill="currentColor"
        height="17"
        stroke="currentColor"
        stroke-width="0"
        viewBox="0 0 16 16"
        width="17"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M1 8a.5.5 0 0 1 .5-.5h11.793l-3.147-3.146a.5.5 0 0 1 .708-.708l4 4a.5.5 0 0 1 0 .708l-4 4a.5.5 0 0 1-.708-.708L13.293 8.5H1.5A.5.5 0 0 1 1 8z"
          fill-rule="evenodd"
        />
      </svg>
    </button>
  </div>
</form>
`;
